src/com/zebrunner/jenkins/jobdsl/factory/pipeline/hook/PullRequestJobFactory.groovy

```diff
@@ -1,40 +1,111 @@
                 configure addHiddenParameter('GITHUB_HOST', '', host)
                 configure addHiddenParameter('GITHUB_ORGANIZATION', '', organization)
-                stringParam('repo', repo, 'Your GitHub repository for scanning')
+                stringParam('pr_number', '', '')
+                stringParam('pr_repository', '', '')
+                stringParam('pr_source_branch', '', '')
+                stringParam('pr_target_branch', '', '')
+                stringParam('pr_action', '', '')
+                stringParam('pr_sha')
             }
+
+            properties {
+                pipelineTriggers {
+                    triggers {
+                        genericTrigger {
+                            genericVariables {
+                                genericVariable {
+                                    key("pr_number")
+                                    value(webHookArgs.prNumber)
+                                }
+                                genericVariable {
+                                    key("pr_repository")
+                                    value(webHookArgs.prRepo)
+                                }
+                                genericVariable {
+                                    key("pr_source_branch")
+                                    value(webHookArgs.prSourceBranch)
+                                }
+                                genericVariable {
+                                    key("pr_target_branch")
+                                    value(webHookArgs.prTargetBranch)
+                                }
+                                genericVariable {
+                                    key("pr_action")
+                                    value(webHookArgs.prAction)
+                                }
+                                genericVariable {
+                                    key("pr_sha")
+                                    value(webHookArgs.prSha)
+                                }
+                            }
+
+                            genericHeaderVariables {
+                                genericHeaderVariable {
+                                    key(webHookArgs.eventName)
+                                    regexpFilter("")
+                                }
+                            }
+
+                            tokenCredentialId("${this.organization}-webhook-token")
+                            printContributedVariables(isDebugActive())
+                            printPostContent(isDebugActive())
+                            silentResponse(false)
+                            regexpFilterText(webHookArgs.prFilterText)
+                            regexpFilterExpression(webHookArgs.prFilterExpression)
+                        }
+                    }
+                }
+            }
+
+            return pipelineJob
         }
-        return pipelineJob
+    }
+
+    protected def isDebugActive() {
```

Comment:
why do we need it? logger has more levels so isDebugActive seems rather strange

Reply:
Update**

Seems like for the sonar we use the same log as in the pipeline, for the generic webhook dsl objects we need a boolean value to print the post and contributed content(i decided to put this in the debug level because its a lot of variables for the info level)

Reply:
see [db31839](https://github.com/zebrunner/pipeline-ce/pull/32/commits/db318394ca4a6f64f593b12d447276211b786f0e)